light/store/db/db.go

```diff
@@ -65,11 +65,11 @@ func setDBKeyLayout(db dbm.DB, lightStore *dbs, dbKeyLayoutVersion string) {
 
 // New returns a Store that wraps any DB (with an optional prefix in case you
 // want to use one DB with many light clients).
-func New(db dbm.DB, prefix string) store.Store {
+func New(db storage.DB, prefix string) store.Store {
```

Comment:
I am not convinced that `DB` interface must be private. This is a public API, but we use `internal` interface here. Say you are an outside developer and want to implement your own database. How are you supposed to know what to pass here? I think the correct way is to expose the `DB` interface and use it in all public API so that developers can pass any DB instance as long as it satisfies the `DB` interface.

Reply:
See #4690